### DISCUSSION TITLE
Debugging go functions with `vercel dev`

### DISCUSSION THREAD

**Discussion by tom-sherman** [vercel/community#4390] [Idea]
Can this be done? Ideally in VSCode 😄 

I've noticed `vercel dev` spawns an ephemeral go process for each request, so I can't attach to a running pid.

**Comment by lissy93** [comment]
I'm wondering the same. Does anyone have any insight?

---

You can get a bit of info running in debug mode (like `npx vercel dev --debug`), but not the logging output from any of the functions

As a workaround, I found a few options which kinda temporarily solve the problem, for me.

### Option 1

Use an external debugging service.
Or write debug logs a temp file. 

### Option 2
Setup a `main.go` as a local web server, to execute serverless functions.
This took me about 20 minutes to get up and running, but I do like that it removes the Vercel vendor lock-in.
